storage/pruning/fileHandling.go

```diff
@@ -0,0 +1,49 @@
+package pruning
+
+import (
+	"io"
+	"os"
+	"strings"
+)
+
+// removeDirectoryIfEmpty will clean the directories after all persisters for one epoch were destroyed
+// the structure is this way :
+// workspace/db/Epoch_X/Shard_Y/DbName
+// we need to remove the last 3 if everything is empty
+func removeDirectoryIfEmpty(path string) {
+	elementsSplitBySeparator := strings.Split(path, string(os.PathSeparator))
+
+	epochDirectory := ""
+	for idx := 0; idx < len(elementsSplitBySeparator)-2; idx++ {
+		epochDirectory += elementsSplitBySeparator[idx] + string(os.PathSeparator)
+	}
+
+	shardDirectory := epochDirectory + elementsSplitBySeparator[len(elementsSplitBySeparator)-2]
+	if isDirectoryEmpty(shardDirectory) {
```

Comment:
you can refactor this part of code !isDirectoryEmpy -> return and after that continue with your code

Reply:
I don't think this is needed because the function doesn't return anything so the code between the brackets will only be executed if the directory is empty. otherwise, no other action is needed

Reply:
you can use return without any value just to go out from function